guides/keras_tuner/getting_started.py

```diff
@@ -2,104 +2,223 @@
         loss="categorical_crossentropy",
         metrics=["accuracy"],
     )
     return model
 
 
+build_model(kt.HyperParameters())
+
+"""
+As shown below, the hyperparameters are actual values. In fact, they are just
+functions returning actual values. For example, `hp.Int()` returns an `int`
+value. Therefore, you can put them into variables, for loops, or if
+conditions.
+"""
+
+hp = kt.HyperParameters()
+print(hp.Int("units", min_value=32, max_value=512, step=32))
+
+"""
+You can also define the hyperparameters in advance and keep your Keras code in
+a separate function.
 """
-## Start the search
 
-Next, let's instantiate a tuner. You should specify the model-building function, the
-name of the objective to optimize (whether to minimize or maximize is
-automatically inferred for built-in metrics), the total number of trials
-(`max_trials`) to test, and the number of models that should be built and fit
-for each trial (`executions_per_trial`).
 
-We use the `overwrite` argument to control whether to overwrite the previous
-results in the same directory or resume the previous search instead.  Here we
-set `overwrite=True` to start a new search and ignore any previous results.
+def call_existing_code(units, activation, dropout, lr):
+    model = keras.Sequential()
+    model.add(layers.Flatten())
+    model.add(layers.Dense(units=units, activation=activation))
+    if dropout:
+        model.add(layers.Dropout(rate=0.25))
+    model.add(layers.Dense(10, activation="softmax"))
+    model.compile(
+        optimizer=keras.optimizers.Adam(learning_rate=lr),
+        loss="categorical_crossentropy",
+        metrics=["accuracy"],
+    )
+    return model
+
+
+def build_model(hp):
+    units = hp.Int("units", min_value=32, max_value=512, step=32)
+    activation = hp.Choice("activation", ["relu", "tanh"])
+    dropout = hp.Boolean("dropout")
+    lr = hp.Float("lr", min_value=1e-4, max_value=1e-2, sampling="log")
+    # call existing model-building code with the hyperparameter values.
+    model = call_existing_code(
+        units=units, activation=activation, dropout=dropout, lr=lr
+    )
+    return model
+
 
-Available tuners are `RandomSearch`, `BayesianOptimization` and `Hyperband`.
+build_model(kt.HyperParameters())
 
-**Note:** the purpose of having multiple executions per trial is to reduce
-results variance and therefore be able to more accurately assess the
-performance of a model. If you want to get results faster, you could set
-`executions_per_trial=1` (single round of training for each model
-configuration).
 """
+Each of the hyperparameters is uniquely identified by its name (the first
+argument). To tune the number of units in different `Dense` layers separately
+as different hyperparameters, we give them different names as `f"units_{i}"`.
+Notably, this is also an example of conditional hyperparameters. The value of
+the hyperparameter specifying the number of `Dense` layers decides how many
+other hyperparameters follows it.
```

Comment:
The sentence "The value of the hyperparameter specifying the number of `Dense` layers decides how many other hyperparameters follows it." isn't entirely clear, please clarify